packages/core/src/Link.js

```diff
@@ -1,29 +1,56 @@
-import styled from 'styled-components'
-import { applyVariations, getPaletteColor, deprecatedColorValue } from './utils'
+import PropTypes from 'prop-types'
+import styled, { css } from 'styled-components'
+import { width, space } from 'styled-system'
+import {
+  mapProps,
+  applyVariations,
+  getPaletteColor,
+  deprecatedColorValue
+} from './utils'
+import { buttonStyles } from '../src/Button'
 
-const Link = styled.a.attrs(props => ({
-  rel: props.target === '_blank' ? 'noopener' : null
-}))`
-  cursor: pointer;
-  text-decoration: none;
-  color: ${getPaletteColor('base')};
+const variations = {
+  fill: css`
+    &:hover {
+      text-decoration: none;
+    }
+  `,
+  link: css`
+    cursor: pointer;
+    text-decoration: none;
+    color: ${getPaletteColor('base')};
 
-  &:hover {
-    color: ${getPaletteColor('dark')};
-    text-decoration: underline;
-  }
+    &:hover {
+      color: ${getPaletteColor('dark')};
+      text-decoration: underline;
+    }
+  `,
+  outline: css`
+    &:hover {
+      text-decoration: none;
+    }
+  `
+}
 
-  ${applyVariations('Link')}
-`
+const Link = mapProps(({ target, ...props }) => ({
+  rel: target === '_blank' ? 'noopener' : null,
+  ...props
+}))(styled.a`
+  ${width} ${space};
+  ${buttonStyles}
```

Comment:
How does everybody feel about export the button styles, including variations and including them here?

Reply:
My only concern is that it makes Link dependent upon another core component, but I'm not sure what implications that might have.

Reply:
True but is it worse to have duplicated code which might become unsynced

Reply:
Good point. I think this would make sense then. I'm generally in favor of DRY; if we run into issues, we could always pull the shared styles into a module and import them from Button/Link, but I doubt we'll need to.